### ISSUE TITLE
Filtering resources feature

### ISSUE THREAD

**Issue by lowcasz** [spring-projects/spring-data-rest#2315]
I think it is very needed feature and will make developing much easier.
My implementation proposition is in the following example which adds "search" parameter into GET method and is very universal.
Example: 
https://www.baeldung.com/rest-api-search-language-spring-data-specifications
https://www.baeldung.com/rest-search-language-spring-jpa-criteria

In addition it can be more universal using other operators, or using url encoded value.
Feauture of functions like "in()" can be added, joining queries with brackets, logical operators, string operators will be great too.

Creating additional methods in interfaces makes a lot of boilrerplate when we are thinking about null values.

In this search parameter we can use DSL, criteria API, JPQL, hibernate filters, or any other stuff. I leave it to discussion what is the best.

To sum up, builded in  filtering feauture would be a great help in the development.

**Comment by mp911de** [comment]
Please either use Querydsl binding customizers or Spring GraphQL if you require more flexibility. We do not intend adding a custom query language on top of the HTTP API.
